skglm/experimental/pdcd_ws.py

```diff
@@ -0,0 +1,225 @@
+import warnings
+
+import numpy as np
+from numpy.linalg import norm
+
+from numba import njit
+from skglm.utils.jit_compilation import compiled_clone
+from sklearn.exceptions import ConvergenceWarning
+
+
+class PDCD_WS:
+    r"""Primal-Dual Coordinate Descent solver with working sets.
+
+    It solves::
+
+        \min_w F(Xw) + G(w) \Leftrightarrow \min_w \max_z <Xw, z> + G(w) - F^*(z)
```

Comment:
It solves (keep the problem in min _w) 

using a primal dual approach on the saddle point problem: 
(put the the min max problem) 

(we write it this way because the two problems are not always equivalent)

Reply:
I deliberately used the saddle point formulation to introduce the conjugate of the datafit and the dual variable  ``z``, which shows up in the scores expression. Doesn't hurt much to keep it?

The equivalence between primal and saddle point formulations holds for convex proximable functions, right?

Reply:
The min problem is equivalent to the min max, yes, but to use the max min and the Fercoq Bianchi solver you need additional assumptions.

I said to keep the saddle point formulation just put it in a new environment below. read my message again.

Reply:
:pray: